stix_shifter_modules/rhacs/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,506 @@
+        """
+        if converted_value.lower() == "true" or (converted_value.isdigit() and converted_value == "1"):
+            boolean_value = bool(True)
+        elif converted_value.lower() == "false" or (converted_value.isdigit() and converted_value == "0"):
+            boolean_value = bool(False)
+        else:
+            raise NotImplementedError('Invalid boolean type input')
+        return boolean_value
+
+    @staticmethod
+    def _format_negate(comparator):
+        """
+        returns negation of input operator
+        :param comparator:str
+        :return str
+        """
+        negate_comparator = {
+            ":>": ":<=",
+            ":>=": ":<",
+            ":<": ":>=",
+            ":<=": ":>",
+            ":": ":!",
+            ":!": ":",
+            ":r/": ":!r/"
+        }
+        return negate_comparator[comparator]
+
+    @staticmethod
+    def _check_time_range_values(converted_timestamp):
+        """
+        checks for valid start and stop time
+        :param converted_timestamp: list
+        """
+        if converted_timestamp[0] > converted_timestamp[1]:
+            raise StartStopQualifierValueException('Start time should be lesser than Stop time')
+
+    @staticmethod
+    def _parse_time_range(qualifier, time_range):
+        """
+        Converts qualifier timestamp to epoch
+        :param qualifier: str
+        :param time_range: int
+        return: list of converted epoch values
+        """
+        try:
+            compile_timestamp_regex = re.compile(START_STOP_PATTERN)
+            if qualifier and compile_timestamp_regex.search(qualifier):
+                time_range_iterator = compile_timestamp_regex.finditer(qualifier)
+                time_range_list = [each.group() for each in time_range_iterator]
+            else:
+                start_time = STOP_TIME - timedelta(minutes=time_range)
+                converted_start_time = start_time.strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3] + 'Z'
+                # limit 3 digit value for millisecond
+                converted_stop_time = STOP_TIME.strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3] + 'Z'
+                time_range_list = [converted_start_time, converted_stop_time]
+        except (KeyError, IndexError, TypeError) as ex:
+            raise ex
+        return time_range_list
+
+    def _add_timestamp_to_query(self, query, qualifier):
+        """
+        adds timestamp filter to rhacs query
+        :param query: str
+        :param qualifier
+        :return str
+        """
+        match = re.search(r'Violation\s*Time:', query)
+        if match:
+            final_query = query
+        else:
+            converted_timestamp = \
+                QueryStringPatternTranslator._parse_time_range(qualifier, self.options["time_range"])
+            QueryStringPatternTranslator._check_time_range_values(converted_timestamp)
+            self.timeframe += converted_timestamp
+            final_query = f'{query}%2BViolation Time:>=' \
```

Comment:
encoded character `%2B` is still hardcoded here.

Reply:
Updated to encode '+' character.